onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -761,8 +761,10 @@ ONNX_MS_OPERATOR_SET_SCHEMA(BiasSoftmax, 1,
                                 .SetDoc(
                                     "Y = softmax(scores + bias)) with simple broadcast on bias. "
                                     "Intended to specialize softmax(scores + additive_mask) commonly found in transformer models.")
-                                .Attr("softmax_axis", "apply softmax to elements for dimensions softmax_axis or higher", AttributeProto::INT, static_cast<int64_t>(1))
-                                .Attr("broadcast_axis", "broadcast bias across input for dimensions broadcast_axis to softmax_axis-1", AttributeProto::INT, static_cast<int64_t>(1))
+                                .Attr("axis", "apply softmax to elements for dimensions axis or higher", AttributeProto::INT, static_cast<int64_t>(1))
+                                .Attr("is_inner_broadcast", "true if broadcast bias across input for dimensions broadcast_axis to axis-1, "
+                                      "otherwise broadcast bias across input for dimensions 0 to broadcast_axis - 1",
+                                      AttributeProto::INT)
```

Comment:
"otherwise broadcast bias across input for dimensions 0 to broadcast_axis - 1" what is broadcast_axis here?

Is this change to the op spec backward compatible? We need to care about backward compatibility if a serialized model can contain this op.

Reply:
I just used "broadcast_axis" here trying to explain the idea. The previous broadcast_axis attribute in the schema is actually useless. The code used it to calculate the broadcasting size, but actually the size is just the input_size/bias_size. This is a design flaw at the beginning. For the backward compacity, I guess for this Op it's OK, it's created by fusion only, not possible to be in any existing graph, and it's for CUDA only, it just doesn't have the CPU kernel so that the KernelDef hash list just doesn't have it's hash there.